packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamTable.tsx

```diff
@@ -46,14 +49,17 @@ export default function ChallengeTeamTable(props: ChallengeTeamTableProps) {
     },
   )
 
-  const allRows: ChallengeTeamRow[] = useMemo(() => {
+  const allRows: RowData[] = useMemo(() => {
     const term = searchValue.toLowerCase()
     const teams =
       teamsList?.list.filter(
         team =>
           team.canPublicJoin ||
           // `canRequestMembership` is true by default if the property is not present
-          team.canRequestMembership !== false,
+          team.canRequestMembership !== false ||
+          openInvitationsToJoinTeams?.some(
+            invitation => invitation.teamId === team.id,
+          ),
```

Comment:
If there is an open invitation for the registered team, then that team can also be shown